File: bigdatakit-example-spark-streaming/src/main/java/com/sogou/bigdatakit/example/streaming/EchoLineProcessor.java
package com.sogou.bigdatakit.example.streaming;

import com.sogou.bigdatakit.streaming.processor.LineProcessor;

/**
 * Created by Tao Li on 11/24/15.
 */
public class EchoLineProcessor implements LineProcessor {

  public void init() {
    System.out.println("init...");
  }

  public void process(String message) {
    System.out.println("process: " + message);

  }

  public void close() {
    System.out.println("close...");
  }
}

File: bigdatakit-example-spark-streaming/src/main/scala/com/sogou/bigdatakit/example/streaming/EchoLineProcessor.scala
package com.sogou.bigdatakit.example.streaming

import com.sogou.bigdatakit.streaming.processor.LineProcessor

/**
  * Created by Tao Li on 6/16/16.
  */
class EchoLineProcessor extends LineProcessor {
  override def beforeBatch(): Unit = {
    println("before batch ...")
  }

  override def afterBatch(): Unit = {
    println("after batch ...")
  }

  override def process(message: String): Unit = {
    println("process: " + message)
  }
}
